### DISCUSSION TITLE
Character victory music plays at wrong time

### DISCUSSION THREAD

**Discussion by potsmugen** [ikemen-engine/Ikemen-GO#1634] [Feature Request] [answered]
### Describe the bug

The character victory music plays at a wrong time when compared to the default screenpack victory music.

(Marginally related but I think victory music could be defined in the def file)

### To Reproduce

1. Add victory music to the screenpack
2. Check timing
3. Add specific victory music to Kung Fu Man
4. Compare the timing

### Expected behavior

Specific music should start at the same time as the win quote screen, like the screenpack music.

### Screenshots / Video

Screenpack music:

https://github.com/ikemen-engine/Ikemen-GO/assets/107247004/c64b130f-6d9a-4dcd-bcb3-7ed47caf1fbe

Character music (from select.def):

https://github.com/ikemen-engine/Ikemen-GO/assets/107247004/182e6c09-f8ae-4546-90cb-b26d6481815e

### Engine Version (or source code date)

10/02/2024

### Operating system

Windows

**Comment by K4thos** [comment, marked as answer]
This was actually intentional. It fulfilled a feature request to implement this type of victory music: https://www.youtube.com/watch?v=vIrfCs_O47Q&t=42s. Perhaps it should be renamed to "win screen" instead of "victory music" to avoid confusion. Or maybe even removed altogether during the Lua rewriting to prevent feature bloat. I believe that the functionality of screenpack should not be expanded upon by default much beyond what Mugen offered, but rather, it should be managed through external expansions. Something like this could be implemented either through an external module or a common state, thanks to the bgm sctrl.

**Reply by potsmugen** [reply]
For the current implementation I guess it's a combination of both "win" (any round end) and "victory" (match end), so maybe it needs a new term. "Pre-victory"? And it should probably be a screenpack parameter rather than character in that case. You'd have to add the music to every single char to achieve that intended effect.

For character victory music in general, I think an elegant implementation of this feature would be defining a victory music in the char's def file. Screenpacks would play it by default in the victory screen, with an option to make the screenpack music override the char music.

My chars have it implemented with PlayBGM but it feels a little hacky.
